statsmodels/duration/survfunc.py

```diff
@@ -60,14 +56,141 @@ def _calc_survfunc_right(time, status, weights=None):
+        v -= 2 * ip[k] * np.cumsum(ip[k] * ra)
+        v += np.cumsum(ip[k]**2 * ra)
+
+        ra = (n - d[k]) * d[k] / n
+        v += np.cumsum(sp0**2 * ra)
+
+        ra = sp0 * d[k] / n
+        v -= 2 * ip[k] * np.cumsum(ra)
+        v += 2 * np.cumsum(ip[k] * ra)
+
+        se.append(np.sqrt(v))
+
+    return ip, se, utime
+
+
+def _checkargs(time, status, freq_weights):
+
+    if len(time) != len(status):
+        raise ValueError("time and status must have the same length")
+
+    if freq_weights is not None and (len(freq_weights) != len(time)):
+        raise ValueError("weights, time and status must have the same length")
+
+
+class CumIncidencefuncRight(object):
+    """
+    Estimation and inference for a cumulative incidence function.
+
+    If J = 1, 2, ... indicates the event type, the cumulative
+    incidence function for cause j is:
+
+    I(t, j) = P(T <= t and J=j)
+
+    Only right censoring is supported.  If frequency weights are provided,
+    the point estimate is returned without a standard error.
+
+    Parameters
+    ----------
+    time : array-like
+        An array of times (censoring times or event times)
+    status : array-like
+        If status >= 1 indicates which event occured at time t.  If
+        status = 0, the subject was censored at time t.
+    title : string
+        Optional title used for plots and summary output.
+    freq_weights : array-like
+        Optional frequency weights
+
+    Attributes
+    ----------
+    times : array-like
+        The distinct times at which the incidence rates are estimated
+    cinc : list of arrays
+        cinc[k-1] contains the estimated cumulative incidence rates
+        for outcome k=1,2,...
+    cinc_se : list of arrays
+        The standard errors for the values in `cinc`.
+
+    References
+    ----------
+    The Stata stcompet procedure:
+        http://www.stata-journal.com/sjpdf.html?articlenum=st0059
+
+    Dinse, G. E. and M. G. Larson. 1986. A note on semi-Markov models
+    for partially censored data. Biometrika 73: 379-386.
+
+    Marubini, E. and M. G. Valsecchi. 1995. Analysing Survival Data
+    from Clinical Trials and Observational Studies. Chichester, UK:
+    John Wiley & Sons.
+    """
+
+    def __init__(self, time, status, title=None, freq_weights=None):
+
+        _checkargs(time, status, freq_weights)
+        time = self.time = np.asarray(time)
+        status = self.status = np.asarray(status)
+        if freq_weights is not None:
+            freq_weights = self.freq_weights = np.asarray(freq_weights)
+        x = _calc_incidence_right(time, status, freq_weights)
+        self.cinc = x[0]
+        self.cinc_se = x[1]
+        self.times = x[2]
+        self.title = "" if not title else title
 
 
 class SurvfuncRight(object):
     """
     Estimation and inference for a survival function.
 
-    Only right censoring is supported.
+    The survival function S(t) = P(T > t) is the probability that an
```

Comment:
just to confirm: Are we using `P(T > t)` or `P(T >= t)`?
I've seen both, but the former is more consistent with standard terminology.